packages/ethers-5/src/client.ts

```diff
@@ -0,0 +1,788 @@
+            await coinbaseProvider.send('wallet_addEthereumChain', [
+              {
+                chainId: `0x${chainId.toString(16)}`,
+                rpcUrls: [NetworkRPCUrls[chainId]],
+                chainName: NetworkNames[chainId],
+                nativeCurrency: {
+                  name: networkCurrenySymbols[chainId],
+                  decimals: 18,
+                  symbol: networkCurrenySymbols[chainId]
+                },
+                blockExplorerUrls: [NetworkBlockExplorerUrls[chainId]],
+                iconUrls: [NetworkImageIds[chainId]]
+              }
+            ])
+          }
+        }
+      }
+    }
+  }
+
+  private syncConnectors(config: ProviderType) {
+    const w3mConnectors: Connector[] = []
+
+    if (config.walletConnect) {
+      const connectorType = ConnectorTypesMap[WALLET_CONNECT_CONNECTOR_ID]
+      if (connectorType) {
+        w3mConnectors.push({
+          id: WALLET_CONNECT_CONNECTOR_ID,
+          explorerId: ConnectorExplorerIds[WALLET_CONNECT_CONNECTOR_ID],
+          imageId: ConnectorImageIds[WALLET_CONNECT_CONNECTOR_ID],
+          name: ConnectorNamesMap[WALLET_CONNECT_CONNECTOR_ID],
+          type: connectorType
+        })
+      }
+    }
+
+    if (config.injected) {
+      const connectorType = ConnectorTypesMap[INJECTED_CONNECTOR_ID]
+      if (connectorType) {
+        w3mConnectors.push({
+          id: INJECTED_CONNECTOR_ID,
+          explorerId: ConnectorExplorerIds[INJECTED_CONNECTOR_ID],
+          imageId: ConnectorImageIds[INJECTED_CONNECTOR_ID],
+          name: ConnectorNamesMap[INJECTED_CONNECTOR_ID],
+          type: connectorType
+        })
+      }
+
+      if (config.coinbase) {
+        w3mConnectors.push({
+          id: COINBASE_CONNECTOR_ID,
+          explorerId: ConnectorExplorerIds[COINBASE_CONNECTOR_ID],
+          imageId: ConnectorImageIds[COINBASE_CONNECTOR_ID],
+          name: ConnectorNamesMap[COINBASE_CONNECTOR_ID],
+          type: 'EXTERNAL'
+        })
+      }
+
+      this.setConnectors(w3mConnectors)
+    }
+  }
+
+  private listenConnectors(enableEIP6963: boolean) {
+    if (typeof window !== 'undefined' && enableEIP6963) {
+      window.addEventListener(EIP6963_ANNOUNCE_EVENT, (event: CustomEventInit<Wallet>) => {
+        if (event.detail) {
+          const { info, provider } = event.detail
+          const eip6963Provider = provider as unknown as ExternalProvider
+          const web3provider = new ethers.providers.Web3Provider(eip6963Provider)
+          const type = ConnectorTypesMap[EIP6963_CONNECTOR_ID]
+
+          if (type) {
+            this.addConnector({
+              id: EIP6963_CONNECTOR_ID,
+              type,
+              imageUrl: info.icon,
+              name: info.name,
+              provider: web3provider,
+              info
+            })
+
+            const name = info.name
+            const eip6963ProviderObj = {
+              name,
```

Comment:
just `name: info.name`, no need to separate it into a constant if it is not used multiple times